File: python/bc-lifehash/src/bc_lifehash/_patterns.py
from __future__ import annotations

from enum import Enum

from ._bit_enumerator import BitEnumerator


class Pattern(Enum):
    SNOWFLAKE = "snowflake"
    PINWHEEL = "pinwheel"
    FIDUCIAL = "fiducial"


def select_pattern(entropy: BitEnumerator, version: Version) -> Pattern:
    from ._lifehash import Version

    if version in (Version.FIDUCIAL, Version.GRAYSCALE_FIDUCIAL):
        return Pattern.FIDUCIAL
    if entropy.next():
        return Pattern.SNOWFLAKE
    return Pattern.PINWHEEL

File: swift/BCLifeHash/Sources/BCLifeHash/Patterns.swift
enum Pattern {
    case snowflake
    case pinwheel
    case fiducial
}

func selectPattern(_ entropy: BitEnumerator, _ version: Version) -> Pattern {
    switch version {
    case .fiducial, .grayscaleFiducial:
        return .fiducial
    default:
        return entropy.next() ? .snowflake : .pinwheel
    }
}
